packages/@react-spectrum/dialog/src/AlertDialog.tsx

```diff
@@ -64,6 +65,32 @@ function AlertDialog(props: SpectrumAlertDialogProps, ref: DOMRef) {
     }
   }
 
+  let [disabledByPending, setDisabledByPending] = useState(false);
+  useEffect(() => {
+    let timeout: ReturnType<typeof setTimeout>;
+    if (pendingAction != null) {
+      // Delay visually disabling other buttons until pending button enters its pending state
+      timeout = setTimeout(() => {
+        setDisabledByPending(true);
+      }, pendingDelay);
+    } else {
+      // Enable dialog buttons when pending action is removed.
+      setDisabledByPending(false);
+    }
+    return () => {
+      // Clean up on unmount or when user clears/changes pendingAction prop before entering pending state.
+      clearTimeout(timeout);
+    };
+  }, [pendingAction]);
+
+  // Prevent Escape from closing the Dialog until pending action is finished
+  let windowRef = useRef(typeof window !== 'undefined' ? window : null);
+  useEvent(windowRef, 'keydown', e => {
+    if (e.key === 'Escape' && pendingAction != null) {
+      e.stopPropagation();
+    }
+  }, {capture: true});
```

Comment:
I thought we decided that users should still be able to cancel even if there is a pending action? I know we had a discussion about it...

Reply:
I guess the user could cancel their in progress request then via onOpenChange then. I'll go ahead and make the cancel button remain non-disabled as well for parity. At that point though it feels like we could just go back and have the other non-pending action remain enabled then?

Reply:
idk, let's discuss this again